lib/ecto/query.ex

```diff
@@ -2667,6 +2667,34 @@ defmodule Ecto.Query do
                  where: l.inserted_at > c.updated_at,
                  preload: [:author, comments: {c, likes: l}]
 
+  ## Choosing Between Preloading with Joins vs. Separate Queries
+
+  Deciding between preloading associations via joins, a single large
+  query, (`preload: [comments: c]`) or separate smaller queries
+  (`preload: [:comments]`) depends on the specific use case.
+  Here are some factors to guide your decision:
+
+  * **Joins reduce database round trips:** By fetching data in a single
+  query, joins can minimize database round trips, potentially reducing
+  overall latency.
+  * **Potential for data duplication:** Joins may lead to duplicated
+  data in the result set, which requires more processing by Ecto
+  and consumes more bandwidth when transmitting the results.
+  * **Increased database load:** Joins can be taxing on the database,
+  increasing the demand on working memory. In some cases, the database
+  may need to write intermediate results to disk, which can slow down
+  query performance.
```

Comment:
This all looks good to me but this part I have some hesitation for a couple reasons:

1. I think this is going a little bit out of Ecto's domain. 
2. In my professional life I've often seen people too scared to use joins when they are appropriate due to not understanding the internals of the database very well. So I'm hesitant to give people information like this who likely will not take the step further to understand when this becomes a real issue vs when the issue is all in their head. Also the stated issues can happen even with non-join queries.

Reply:
Hey, thanks for the comments! I definitely don't want people to be afraid of joins in general. What I want to discourage is "joins just for the sake of preloading" and even that not completely.

To me this is one of the bigger reasons why I think that overusing joins for preloads is a problem hence I wanted to include it.

Do you have a solution in mind? Get rid off the point? Soften it up? Link to some other source, as you're right and it's complex and we can't aim to capture all the nuance involved here.

Reply:
I would personally leave out anything about database internals all together. I think it's the kind of topic that you need to do a lot of digging and self discovery and not something that you can get from a single source. 

But I am not so opposed that I would insist on leaving it out if both you and Jose want it in. I just personally find it more misleading than helpful.

Reply:
In SQLite it is not recommended to preload with joins and it is actively encouraged to do N+1 queries as it is more memory efficient. The underlying adapter implementation _could_ clarify these.

Reply:
@greg-rychlewski to me it's "I'm just a contributor, you know the project, audience, needs & style better than me" so I'm always more inclined to do what the maintainers want :) And as @warmwaffles points out (thanks, didn't know that!) it can vary wildly between database implementations. That said, taking PostgreSQL as the standard for ecto in the elixir community doesn't sound too bad.

Should I remove the point or should we wait for some others to chime in? I don't think there is a fully DB agnostic way to make that point, although I think it's fair to assume that for most DBs a join is an amount of more work vs. a single query.

Reply:
I would wait to see what Jose says. He is definitely more in tune with what would be best for a wider audience.

Reply:
Yea @PragTob it's counter intuitive for SQLite, but that's mainly because it's an embedded database so you don't have to pay for that TCP round trip retrieving data. You are limited on disk speed access. SQLite will also consume way more memory when you try to preload with joins. Just fun things to think about.

Reply:
Agreed that it is hard for us to estimate or argue the database load side. :)

Reply:
Alright, thank you everyone - I removed the section about database load!